pkg/storage/storagewrappers/combinedtuplereader.go

```diff
@@ -29,11 +29,19 @@ type CombinedTupleReader struct {
 var _ storage.RelationshipTupleReader = (*CombinedTupleReader)(nil)
 
 // filterTuples filters out the tuples in the provided slice by removing any tuples in the slice
-// that don't match the object and relation provided in the filterKey.
-func filterTuples(tuples []*openfgav1.TupleKey, targetObject, targetRelation string) []*openfgav1.Tuple {
+// that don't match the object, relation or user provided in the filterKey.
+//
+//nolint:unparam
+func filterTuples(tuples []*openfgav1.TupleKey, targetObject, targetRelation string, optionalParams ...string) []*openfgav1.Tuple {
```

Comment:
Can you update `ReadUserTuple`? i.e. instead of this

```go
filteredContextualTuples := filterTuples(c.contextualTuples, tk.GetObject(), tk.GetRelation())
	for _, t := range filteredContextualTuples {
		if t.GetKey().GetUser() == tk.GetUser() {
			return t, nil
		}
	}
```

you can do

```go
filteredContextualTuples := filterTuples(c.contextualTuples, tk.GetObject(), tk.GetRelation(), tk.GetUser())

```

😉